src/traits.rs

```diff
@@ -884,33 +884,21 @@ impl<'a> FindToken<u8> for &'a [u8] {
   }
 }
 
-impl<'a> FindToken<u8> for &'a str {
-  fn find_token(&self, token: u8) -> bool {
-    self.as_bytes().find_token(token)
-  }
-}
-
 impl<'a, 'b> FindToken<&'a u8> for &'b [u8] {
   fn find_token(&self, token: &u8) -> bool {
     self.find_token(*token)
   }
 }
 
-impl<'a, 'b> FindToken<&'a u8> for &'b str {
-  fn find_token(&self, token: &u8) -> bool {
-    self.as_bytes().find_token(token)
-  }
-}
-
```

Comment:
that breaking change

Reply:
Yes - I'm a little concerned by this as a breaking change.

I needed this to solve:
```
error[E0277]: the trait bound `&str: FindToken<<I as InputIter>::Item>` is not satisfied
    --> src/number/complete.rs:1377:5
     |
1377 |     one_of("0123456789abcdefABCDEF"),
     |     ^^^^^^ the trait `FindToken<<I as InputIter>::Item>` is not implemented for `&str`
     |
```

I believe the issue is that the output parser of `one_of` takes an `I` where `I::Item: AsChar`, but it can't figure out how to deal with that with the input as a `&str`. Thus can't generalize the input of `hex_u32` here while still being able to call `one_of`. I solved by having all `FindToken` on `&str` use the `as_chars()` implementation, but it is definitely a breaking change.

Can you help think of a better way which isn't a breaking change (or alternately, of some way to justify the breaking change)?

Reply:
well your change is not bad could be nice for v8 but now you could just call `as_char` on the item of the iterator

Reply:
I was able to eliminate this change w/ a bit of refactoring and using the as_char suggestion. I had to give up on using preexisting `one_of` method, but it turned out ok. Thanks for pushing on this.